### PR TITLE
Feature/#194 normalize stippling factor

### PR DESCRIPTION
**Pull request by Ish-rivera**

### CODE REVIEW

File: sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java [hunk 1]
```diff
@@ -287,17 +287,28 @@ private void renderShapeParser(List<IFeature> featureList,
...
                 currentStrokeStyle = new GeoStrokeStyle();
-                currentStrokeStyle.setStrokeWidth((renderStrokeStyle == null)? 3: renderStrokeStyle.getStrokeWidth());
-                geoLineColor = new EmpGeoColor((double) lineColor.getAlpha() / 255.0, lineColor.getRed(), lineColor.getGreen(), lineColor.getBlue());
-                currentStrokeStyle.setStrokeColor(geoLineColor);
+                currentStrokeStyle.setStrokeWidth((symbolStrokeStyle == null)? 3: symbolStrokeStyle.getStrokeWidth());
+                newLineColor = currentStrokeStyle.getStrokeColor();
+                newLineColor.setAlpha((double) lineColor.getAlpha() / 255.0);
+                newLineColor.setRed(lineColor.getRed());
+                newLineColor.setGreen(lineColor.getGreen());
+                newLineColor.setBlue(lineColor.getBlue());
```
Comment (empbarney) [inline sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java]: I suggest creating a helper method in EmpStyles helper class. This newLineColor is very confusing, all this code is doing is updating is currentStrokeStyle. Createing thos helper functions will remove this confusion.
Also Some comments on what is happening wrt what is returned by the renderer and what is in the feature will help.

File: sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java [hunk 2]
```diff
@@ -312,9 +323,12 @@ private void renderShapeParser(List<IFeature> featureList,
...
             fillColor = shapeInfo.getFillColor();
             if (fillColor != null) {
-                geoFillColor = new EmpGeoColor((double) fillColor.getAlpha() / 255.0, fillColor.getRed(), fillColor.getGreen(), fillColor.getBlue());
                 currentFillStyle = new GeoFillStyle();
-                currentFillStyle.setFillColor(geoFillColor);
+                geoFillColor =  currentFillStyle.getFillColor();
+                geoFillColor.setAlpha((double) fillColor.getAlpha() / 255.0);
+                geoFillColor.setRed(fillColor.getRed());
+                geoFillColor.setGreen(fillColor.getGreen());
+                geoFillColor.setBlue(fillColor.getBlue());
```
Comment (empbarney) [inline sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java]: Similar comments as newLineColor.

File: sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java [hunk 3]
```diff
@@ -55,6 +56,9 @@
     private static final String TAG = MilStdRenderer.class.getSimpleName();
     private static boolean debugMe = false; // Added to debug Issue #92. Will remove when all problems related to it are resolved.
 
+    // The pixel density of the device display.
+    private static final double STIPPLE_FACTOR_DISPLAY_DENSITY_MODIFIER = (double) Resources.getSystem().getDisplayMetrics().densityDpi / 96.0;
+
```
Comment (empbarney) [inline sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java]: Some explanation of what and why would help.

File: sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java [hunk 4]
```diff
@@ -704,25 +724,39 @@ private void renderBasicShapeParser(List<IFeature> featureList,
...
-                } else {
-                    currentStrokeStyle.setStrokeColor(storageManager.getSelectedStrokeStyle(mapInstance).getStrokeColor());
+                IGeoStrokeStyle selectedStrokeStyle = storageManager.getSelectedStrokeStyle(mapInstance);
+
+                if (null != selectedStrokeStyle.getStrokeColor()) {
+                    IGeoColor selectedColor = selectedStrokeStyle.getStrokeColor();
+                    currentStrokeStyle.getStrokeColor().setAlpha(selectedColor.getAlpha());
+                    currentStrokeStyle.getStrokeColor().setRed(selectedColor.getRed());
+                    currentStrokeStyle.getStrokeColor().setGreen(selectedColor.getGreen());
+                    currentStrokeStyle.getStrokeColor().setBlue(selectedColor.getBlue());
```
Comment (empbarney) [inline sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java]: Same comments as earlier wrt helper class and helper method.

File: sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java (hunk 1 above)
Comment (Ish-rivera) [reply sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java]: done.

File: sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java (hunk 2 above)
Comment (Ish-rivera) [reply sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java]: done

File: sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java (hunk 3 above)
Comment (Ish-rivera) [reply sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java]: Moved and documented.

File: sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java (hunk 4 above)
Comment (Ish-rivera) [reply sdk/sdk-core/aar/src/main/java/mil/emp3/core/utils/MilStdRenderer.java]: done

**Review by empbarney — APPROVED** [review]

**Pull request merged**